modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,408 @@
+ */
+function wp_get_dominant_color( $id, $default_color = 'eee' ) {
+
+	add_filter( 'wp_image_editors', 'wp_dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$dominant_color = wp_get_image_editor( $file )->get_dominant_color( $default_color );
+
+	remove_filter( 'wp_image_editors', 'wp_dominant_color_set_image_editors' );
+
+	return $dominant_color;
+
+}
+
+
+/**
+ * Works out if color has transparency
+ *
+ * @param integer $id the attachment id.
+ *
+ * @return bool
+ */
+function wp_get_has_transparency( $id ) {
+
+	add_filter( 'wp_image_editors', 'wp_dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$editor = wp_get_image_editor( $file );
+	if ( is_wp_error( $editor ) ) {
+
+		return true; // safer to set to trans than not.
+	}
+	$has_transparency = wp_get_image_editor( $file )->get_has_transparency();
+
+	remove_filter( 'wp_image_editors', 'wp_dominant_color_set_image_editors' );
+
+	return $has_transparency;
+}
+
+
+/**
+ * Works out if the color is dark or light.
+ *
+ * @param string $hex color in hex.
+ *
+ * @return bool
+ */
+function wp_color_is_light( $hex ) {
+	$hex       = str_replace( '#', '', $hex );
+	$r         = ( hexdec( substr( $hex, 0, 2 ) ) / 255 );
+	$g         = ( hexdec( substr( $hex, 2, 2 ) ) / 255 );
+	$b         = ( hexdec( substr( $hex, 4, 2 ) ) / 255 );
+	$lightness = round( ( ( ( max( $r, $g, $b ) + min( $r, $g, $b ) ) / 2 ) * 100 ) );
+
+	return $lightness >= 50;
+}
+
+
+/**
+ * Get dominant color and adds it to the image meta and saves it for next time.
+ *
+ * @param int   $attachment_id the attachment id.
+ * @param array $image_meta the current image meta.
+ *
+ * @return array the updated image meta.
+ */
+function wp_back_fill_dominant_color( $attachment_id, $image_meta ) {
+
+	$dominant_color = wp_get_dominant_color( $attachment_id );
+	if ( $dominant_color ) {
+		$image_meta['dominant_color']   = $dominant_color;
+		$image_meta['has_transparency'] = wp_get_has_transparency( $attachment_id );
+		$image_meta['is_light']         = wp_color_is_light( $dominant_color );
+		wp_update_attachment_metadata( $attachment_id, $image_meta );
+	}
+
+	return $image_meta;
+}
+
+/**
+ * Get dominant color and saves it to the image meta for next time.
+ * called by cron task
+ *
+ * @param int $attachment_id the attachment id.
+ *
+ * @return void.
+ */
+function wp_cron_back_fill_dominant_color( $attachment_id ) {
+
+	$dominant_color = wp_get_dominant_color( $attachment_id );
+	if ( $dominant_color ) {
+		$image_meta                     = wp_get_attachment_metadata( $attachment_id );
+		$image_meta['dominant_color']   = $dominant_color;
+		$image_meta['has_transparency'] = wp_get_has_transparency( $attachment_id );
+		$image_meta['is_light']         = wp_color_is_light( $dominant_color );
+		wp_update_attachment_metadata( $attachment_id, $image_meta );
+	}
+}
+add_action( 'wp_back_fill_dominant_color', 'wp_cron_back_fill_dominant_color', 10, 2 );
```

Comment:
See my other comment, can we remove this for now and instead explore that via a follow-up PR? It makes it more complex and IMO isn't necessary for a first MVP.

Reply:
@felixarntz  Should I add a second module to enable this? or create a pull request to update this module?

Reply:
https://github.com/WordPress/performance/pull/303 this is the PR to update the current module